cluster-master.js

```diff
@@ -307,15 +315,23 @@ function forkListener () {
 
     worker.on("disconnect", function () {
       debug("Worker "+id+" disconnect")
-      // give it 1 second to shut down gracefully, or kill
-      disconnectTimer = setTimeout(function () {
-        debug("Worker "+id+", forcefully killing")
-        worker.process.kill("SIGKILL")
-      }, gracefulShutdownDelayMs)
+      startDisconnectTimer(worker, id)
     })
   })
 }
 
+function startDisconnectTimer(worker, id) {
+  clearTimeout(worker.disconnectTimer)
```

Comment:
More of a JavaScript semantics question - it looks like we set `worker.disonnectTimer` in this function - I take it it's fine to pass `undefined` to `clearTimeout`?

Reply:
According to https://developer.mozilla.org/en-US/docs/Web/API/clearTimeout, `clearTimeout`:

> Passing an invalid ID to clearTimeout() silently does nothing; no exception is thrown.